File: Demos/Datasnap/Server/proxy/java_blackberry/com/embarcadero/javablackberry/TDBXBooleanValue.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javablackberry;

/**
 * 
 * Wrap the Boolean type and allows it to be null
 *
 */

public class TDBXBooleanValue extends DBXValue {

	protected boolean ValueNull = false;
	private boolean DBXBooleanValue;
	
	public TDBXBooleanValue() {
		super();
		setDBXType(DBXDataTypes.BooleanType);
	}
	
	public void setNull() {
		ValueNull = true;
		DBXBooleanValue = false;
	}

	public boolean isNull() {
		return ValueNull;
	}

	public void SetAsBoolean(boolean Value) throws DBXException {		
		DBXBooleanValue = Value;
		ValueNull = false;
	}
	
	public boolean GetAsBoolean() throws DBXException {
		return DBXBooleanValue;
	}

}

File: Demos/Datasnap/Server/proxy/objectivec_ios81/Proxy/TDBXBooleanValue.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TDBXBooleanValue.h"


@implementation TDBXBooleanValue
-(id)init{
	self = [super init];
	if (self) {
		[self setDBXType:BooleanType];
	}
	return self;
}
-(void) SetNull {
	ValueNull = YES;
	DBXBoolValue = NO;
}

-(bool) isNull {
	return ValueNull;
}

-(void) SetAsBoolean:(bool) value{
	ValueNull = NO;
	DBXBoolValue = value;
}

-(bool) GetAsBoolean {
	return DBXBoolValue;
}

@end
